### ISSUE TITLE
NIP 13 POW for other events

### ISSUE THREAD

**Issue by thesimplekid** [0xtlt/nostr_rust#14]
The current implementation of NIP13 is only implemented for text notes an requires that a new POW function be written for each type of event, for example `fn publish_pow_text_note` for `fn publish_text_note`. To avoid this duplication `fn to_event` could be changed to also take a difficulty target argument, `Option<u32>`, and create an event with or without proof of work based on the option. 
This would require functions that call `fn to_event`, `fn set_metadata`, `fn add_recommended_relay` `fn set_contact_list` etc, to also be changed to accept this argument. 
Curious to know your thoughts on which route to take as you may not want these existing functions changed.  Thanks

**Comment by 0xtlt** [comment]
Yes It's a good idea :) u16 could be the good type

If you want to work on it tell me!

**Comment by thesimplekid** [comment]
Yes I can work on it change `fn to_event` and functions that call it to accept a u16.
